### PR TITLE
Fix twitter (x) posts for v2 api

### PR DESCRIPTION
**Pull request by jackrosa183**

### CODE REVIEW

File: lib/tilex/notifications/notifiers/twitter.ex [hunk 1]
```diff
@@ -1,7 +1,9 @@
 defmodule Tilex.Notifications.Notifiers.Twitter do
+  alias OAuther
   alias Tilex.Blog.Developer
 
   use Tilex.Notifications.Notifier
+  require Logger
```
Comment (avogel3) [inline lib/tilex/notifications/notifiers/twitter.ex:6]: what's this used for?

File: lib/tilex/notifications/notifiers/twitter.ex [hunk 2]
```diff
@@ -16,7 +18,40 @@ defmodule Tilex.Notifications.Notifiers.Twitter do
     :ok
   end
 
-  def send_tweet(text) do
-    ExTwitter.update(text)
+  def send_tweet(message) do
+    url = "https://api.x.com/2/tweets"
```
Comment (avogel3) [inline lib/tilex/notifications/notifiers/twitter.ex:22]: could we hoist this up to something like this -
```
@tweet_url "https://api.x.com/2/tweets"
```

**Comment by avogel3** [comment]
A few small things to clean up: we should remove the `extwitter` dependency since we are no longer using it -and- pull/repurpose the it's config in `config/config.exs`. You might be able to adjust it a bit so that you can use it in the `oauther_credentials` function.

File: lib/tilex/notifications/notifiers/twitter.ex [hunk 3]
```diff
@@ -16,7 +18,40 @@ defmodule Tilex.Notifications.Notifiers.Twitter do
...
+
+    params = %{
+      "text" => message
+    }
+
+    headers =
+      oauth_headers("post", url) ++
+        [{"Content-Type", "application/json"}]
```
Comment (avogel3) [inline lib/tilex/notifications/notifiers/twitter.ex:30]: did it not work without the content type header?

Comment (jackrosa183) [reply lib/tilex/notifications/notifiers/twitter.ex:30]: I just tried without it and it does work 👍

Comment (avogel3) [reply lib/tilex/notifications/notifiers/twitter.ex:30]: awesome!

File: lib/tilex/notifications/notifiers/twitter.ex (hunk 2 above)
Comment (jackrosa183) [reply lib/tilex/notifications/notifiers/twitter.ex]: yes!

File: lib/tilex/notifications/notifiers/twitter.ex (hunk 1 above)
Comment (jackrosa183) [reply lib/tilex/notifications/notifiers/twitter.ex]: Oops that was for some debugging i forgot to remove it.

**Review by avogel3 — APPROVED** [review]
LGTM 👍

**Pull request merged**